src/nlsolve/anderson.jl

```diff
@@ -135,22 +136,27 @@ function (S::NLAnderson{true})(integrator)
       mul!(ztmp, mass_matrix, z)
       @. z₊ -= ztmp
     end
-    gs[:,1] .= vec(z₊)
+    gs[1] .= vec(z₊)
 
     mk = min(S.n, iter-1)
-    residuals[:,1:mk] .= (gs[:,2:mk+1] .- zs[:,2:mk+1]) .- (gs[:,1] - zs[:,1])
-    alphas[1:mk] .= residuals[:,1:mk] \ (zs[:,1] .- gs[:,1])
+    ztmp = (gs[1] - zs[1])
```

Comment:
get rid of this allocation?

Reply:
Would replacing `=` with `.=` serve the purpose?

Reply:
If `ztmp` already exists.